tb_plugin/torch_tb_profiler/profiler/data.py

```diff
@@ -171,6 +179,14 @@ def process(self):
         self.blocks_per_sm_count = gpu_metrics_parser.blocks_per_sm_count
         self.occupancy_count = gpu_metrics_parser.occupancy_count
 
+        memory_parser = MemoryParser(module_parser.tid2tree)
+        memory_parser.parse_events(self.events)
+        self.memory_state_cpu = memory_parser.get_memory_stats(DeviceType.CPU, -1)
+        if self.gpu_ids:
+            self.memory_state_cuda = {}
+            for gpu_id in self.gpu_ids:
+                self.memory_state_cuda[gpu_id] = memory_parser.get_memory_stats(DeviceType.CUDA, gpu_id)
```

Comment:
Traversing the tree multiple times may have space to increase perf. overhead. Can comment or record somewhere for future optimization.